application/plugin_base/item_list.py

```diff
@@ -1647,6 +1665,19 @@ def destroy_providers(self):
 		for path, provider in self._providers.items():
 			provider.release_archive_handle()
 
+	def create_intermediate_archive_provider(self, path=None):
```

Comment:
I have updated my commits to remove the recursive method and move the code
to the one place that needs the logic (I think). Feedback welcome!
Christian

On Mon, Feb 9, 2015 at 4:00 PM, Mladen Mijatov <notifications@github.com>
wrote:

> In application/plugin_base/item_list.py
> <https://github.com/MeanEYE/Sunflower/pull/65#discussion_r24338353>:
>
> > @@ -1647,6 +1665,19 @@ def destroy_providers(self):
> >  		for path, provider in self._providers.items():
> >  			provider.release_archive_handle()
> >
> > +	def create_intermediate_archive_provider(self, path=None):
>
> I understand the purpose, but am not really liking the way it works. It's
> a single purpose method for only specific use cases. There has to be a
> better way. Issue can be partially mitigated by getting provider from the
> opposite object, from which we are inheriting path anyway.
>
> That leaves issue of restoring path when restarting Sunflower, but I feel
> that should be a separate function.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/MeanEYE/Sunflower/pull/65/files#r24338353>.
>